lib/synapse.rb

```diff
@@ -57,11 +57,14 @@ def run
         if @config_updated
           @config_updated = false
           @config_generators.each do |config_generator|
-            log.info "synapse: regenerating #{config_generator.name} config"
+            log.info "synapse: configuring #{config_generator.name}"
             config_generator.update_config(@service_watchers)
           end
         else
```

Comment:
Would it be better to kill off the `else` and instead *always* execute the `sleep` and `config_generator.update(...)` calls?  The way the virtual time doesn't stall when an `config_generator.update_config(...)` happens.  This might also cause us to more effectively batch the HAProxy socket operations (with the downside being that there could be up to 1s latency, but I think that's OK?)